src/clj/puppetlabs/trapperkeeper/services/watcher/filesystem_watch_core.clj

```diff
@@ -40,11 +40,11 @@
         changed-path (when-not (= :unknown kind)
                        (.context event))
         full-path (when-not (= :unknown kind)
-                    (-> watched-path (.resolve changed-path) fs/file))]
+                    (-> watched-path (.resolve changed-path) (.toFile)))]
     {:type kind
      :count (.count event)
-     :watched-path (fs/file watched-path)
-     :changed-path (fs/file changed-path)
+     :watched-path (.toFile watched-path)
+     :changed-path (when changed-path (.toFile changed-path))
```

Comment:
We create the (absolute) `:full-path` by resolving the (relative) `:changed-path` relative to the (absolute) `:watched-path`, when we call `fs/file` on the `:changed-path` it resolves it to an absolute path relative to the processes current working directory.

Here's an mock up of the before and after data:
before this change:
```
{:type :create
 :count 1
 :watched-path java.io.File["/var/folders/4l/single-path-test/"]
 :changed-path java.io.File["/Users/justin/code/tk-fs-watcher/first-file"]
 :full-path java.io.File["/var/folders/4l/single-path-test/first-file"]}
```
after this change:
```
{:type :create
 :count 1
 :watched-path java.io.File["/var/folders/4l/single-path-test/"]
 :changed-path java.io.File["first-file"]
 :full-path java.io.File["/var/folders/4l/single-path-test/first-file"]}
```